### DISCUSSION TITLE
Installing on Linux after compiling from source

### DISCUSSION THREAD

**Discussion by kindrowboat** [wezterm/wezterm#2128] [Q&A]
This question is more of a sanity/correctness check. I'm not every familair with the rust build toolchain yet.

I was building WezTerm from source for a Raspberry Pi following [these instructions](https://wezfurlong.org/wezterm/install/source.html) and everything compiled and is running correctly (yay!) To install it, I ran 

```bash
cd target/release/
sudo install wezterm wezterm-gui wezterm-mux-server strip-ansi-escapes /usr/local/bin
```

Everything seems to be working fine. Is that the right way to go about it? Should I be putting any of the other files (e.g. the libwezterm_dynamic_derive.so) somewhere? Would it make sense to make a PR to add this to the docs? Thanks!

**Comment by wez** [comment]
https://github.com/wez/wezterm/blob/main/ci/deploy.sh encodes the installation logic for various platforms.

What I think would be good is to refactor the linux portions so that there is a generic "create an install tree under a target prefix" script, and have the other places call that:

eg: `sudo ci/install.sh /usr/local`

**Comment by tjex** [comment]
For any who have also found their way here with issues, I wrote a little post about this after I got it working.

https://tjex.net/hacks/installing-wezterm-on-a-raspberry-pi-or-linux-server/

**Comment by noahmbright** [comment]
I'm also new to the Rust toolchain and I'm having some trouble understanding how to install. I'm on MacOS, and I can build from source and run just fine. I tried running `./ci/deploy` from the root of the repo, but `which wezterm` gave no result. In `deploy` I see most distros have ~10 commands like `install -Dsm755 -t pkg/debian/usr/bin target/release/wezterm-mux-server`, which mostly make sense to me, but I don't really understand what each one does individually. Is there a canonical file structure somewhere we can install those ~10 files to? If so we could start the refactor that was mentioned earlier
